src/black/strings.py

```diff
@@ -236,3 +246,31 @@ def normalize_string_quotes(s: str) -> str:
         return s  # Prefer double quotes
 
     return f"{prefix}{new_quote}{new_body}{new_quote}"
+
+
+def normalize_unicode_escape_sequences(leaf: Leaf) -> None:
+    """Replace hex codes in Unicode escape sequences with lowercase representation."""
+    text = leaf.value
+    prefix = get_string_prefix(text)
+
+    def replace(m: Match[AnyStr]) -> str:
```

Comment:
```suggestion
    def replace(m: Match[str]) -> str:
```
Seems like you only handle strs here, not bytes. You can also remove a bunch of the str() calls below.